fs-bucket-listobjects.go

```diff
@@ -120,41 +113,48 @@ func (fs Filesystem) ListObjects(bucket, prefix, marker, delimiter string, maxKe
 	// popTreeWalker returns nil if the call to ListObject is done for the first time.
 	// On further calls to ListObjects to retrive more objects within the timeout period,
 	// popTreeWalker returns the channel from which rest of the objects can be retrieved.
-	walker := fs.popTreeWalker(listObjectParams{bucket, delimiter, marker, prefix})
-	if walker == nil {
-		walker = startTreeWalker(fs.path, bucket, filepath.FromSlash(prefix), filepath.FromSlash(marker), recursive)
+	walkNotify := fs.popTreeWalkNotify(listObjectParams{bucket, delimiter, marker, prefix})
+	if walkNotify == nil {
+		walkNotify = startTreeWalk(fs.path, bucket, filepath.FromSlash(prefix), filepath.FromSlash(marker), recursive)
 	}
```

Comment:
walkNotify -> treeWalkCh
It should be a channel. Timeout should be hidden inside the go routine closure.